### PR TITLE
Make s3 bucket lookup type configurable

### PR DESCRIPTION
**Pull request by damnever**
**What this PR does**:

**Which issue(s) this PR fixes**:
Some providers may only support one of those bucket lookup styles.

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 * [FEATURE] Compactor: Added `-compactor.block-files-concurrency` allowing to configure number of go routines for download/upload block files during compaction. #4784
 * [ENHANCEMENT] Querier/Ruler: Retry store-gateway in case of unexpected failure, instead of failing the query. #4532
 * [FEATURE] Compactor: Added -compactor.blocks-fetch-concurrency` allowing to configure number of go routines for blocks during compaction. #4787
+* [FEATURE] Storage/Bucket: Added `-*.s3.bucket-lookup-type` allowing to configure the s3 bucket lookup type. #4787
```
Comment (danielblando) [inline CHANGELOG.md:7]: Need to update the PR id. Seems to be a copy/paste problem

**Comment by danielblando** [comment]
@damnever could you fix the lint?

**Comment by damnever** [comment]
@danielblando  done

File: pkg/storage/bucket/s3/config.go [hunk 2]
```diff
@@ -95,6 +106,19 @@ func (cfg *Config) Validate() error {
 	return nil
 }
 
+func (cfg *Config) bucketLookupType() s3.BucketLookupType {
```
Comment (danielblando) [inline pkg/storage/bucket/s3/config.go:109]: What is your opinion on doing this the same as "BuildThanosConfig()"?

Returning an error when it goes to default which can also return the AutoLookup. The caller can decide what to do. I think at least we are not swallowing any issues.

Comment (damnever) [reply pkg/storage/bucket/s3/config.go:109]: I think `Validate` is sufficient for our case. However, returning an error is better for other use.

**Review by danielblando — APPROVED** [review]
Nice, thanks

**Comment by alanprot** [comment]
Seems there are some tests failing now.

**Review by alanprot — APPROVED** [review]

**Review by alvinlin123 — APPROVED** [review]

**Comment by alvinlin123** [comment]
LGTM, thanks for the contribution!

**Pull request merged**